src/intrprtr.c

```diff
@@ -4682,7 +4682,22 @@ void            IntrInfoBegin( void )
 
 }
 
-Obj             InfoDecision;
+static Obj InfoDecision;
+static Obj InfoDecisionFast;
+static Obj IsInfoClassListRep;
+
+Obj InfoCheckLevel(Obj selectors, Obj level)
+{
+    if (IS_POS_INTOBJ(level) &&
+        CALL_1ARGS(IsInfoClassListRep, selectors) == True) {
+        Obj index = ELM_PLIST(selectors, 1);
+        return CALL_3ARGS(InfoDecisionFast, index, selectors, level);
```

Comment:
If `InfoDecisionFast` is only meant to be used by this fast path, why not implement it in C, too? Would that perhaps also improve performance (I have no idea where the bottlenecks really are; perhaps you can document your benchmarks as comments on the PR?